llvm/lib/Target/NVPTX/NVPTX.td

```diff
@@ -33,20 +33,61 @@ class FeaturePTX<int version>:
    SubtargetFeature<"ptx"# version, "PTXVersion",
                     "" # version,
                     "Use PTX version " # version>;
-
+//
+// NVPTX Architecture Hierarchy and Ordering:
+// 
+// Family: 2/3/5/6/7/8/9/10/12 (Follows Onion model, older family is compatible with newer family)
+// Arch: 2*/3*/5*/6*/7*/8*/9*/10*/12*
+//
+// Family-specific: F*f : F*f > F* =>
+// + The plain base architecture is compatible with the family-specific architecture
+//   (e.g. sm_100 compatible with >= sm_100*f*)
```

Comment:
The terms used should be defined before use. E.g. describe that architecture consists of
`sm_` prefix, 
1+ digits of major revision number,
1 digit of minor revision number
optional suffix specifying arch-accelerated of 'family' variant.

Only then proceed describing various ordering properties between them.